src/main/java/lotto/Model/Game.java

```diff
@@ -29,34 +39,59 @@ public List<Ticket> allTickets() {
         return tickets;
     }
 
-    public HashMap<Integer, Integer> calculateResult(List<Integer> winnerNumber) {
-        HashMap<Integer, Integer> result = new HashMap<>();
-        for (int i = WINNER_MATCH_COUNT_MIN; i <= WINNER_MATCH_COUNT_MAX; i++) {
-            result.put(i, 0);
-        }
+    public HashMap<Integer, Integer> calculateResult() {
+        HashMap<Integer, Integer> result = resultFormat();
 
         for (Ticket ticket : tickets) {
-            int count = countMatches(ticket, winnerNumber);
+            int count = countMatches(ticket);
             result.put(count, result.getOrDefault(count, 0) + 1);
         }
 
+        for (int i = 0; i < WINNER_MATCH_COUNT_MIN; i++) {
+            result.remove(i);
+        }
+
         return result;
     }
 
-    private int countMatches(Ticket ticket, List<Integer> winnerNumber) {
+    private HashMap<Integer, Integer> resultFormat() {
+        HashMap<Integer, Integer> result = new HashMap<>();
+        for (int i = WINNER_MATCH_COUNT_MIN; i <= WINNER_MATCH_COUNT_MAX; i++) {
+            result.put(i, 0);
+        }
+        result.put(BONUS_CATEGORY, 0);
+
+        return result;
+    }
+
+    private int countMatches(Ticket ticket) {
         int count = 0;
         TicketNumber ticketNumber = ticket.numbers();
         for (Integer number : ticketNumber.numbers()) {
             count += addIfMatch(winnerNumber, number);
+            count = ifMatchBonus(ticket, count);
         }
         return count;
     }
 
-    private int addIfMatch(List<Integer> numbers, int number){
-        if(numbers.contains(number)){
+    private int addIfMatch(List<Integer> numbers, int number) {
+        if (numbers.contains(number)) {
             return 1;
         }
         return 0;
     }
 
+    private int ifMatchBonus(Ticket ticket, int count) {
+        if (BONUS_CANDIDATE_COUNT != count) {
+            return count;
+        }
+
+        TicketNumber ticketNumber = ticket.numbers();
+        if (ticketNumber.contain(bonusNumber)) {
+            return BONUS_CATEGORY;
+        }
+
+        return count;
+    }
```

Comment:
```suggestion
    public Result calculateResult(List<Integer> winners, int bonus) {
        for (Ticket ticket : tickets) {
            int matches = ticket.countMatches(winners);
            result.increase(matches, ticket.contain(bonus));
        }
        return result;
    }
```

- Game 객체에서 당첨번호 포함 여부, 보너스 번호 포함 여부, 등수 판단, 게임 결과 응답값 바인딩 등의 모든 역할을 가지기 보다 각 객체에게 위임하는 건 어떨까요
- Result 객체를 도출하는건 어떨까요? 그리고 BONUS_CATEGORY를 Map의 특정 key로 관리하는 건 히스토리를 파악하기 어렵고(주석을 남겨야 이해가 되겠네요), 추후 사이드이펙트를 발생할 여지가 있어요.